### PR TITLE
Fix check view navigation

### PR DESCRIPTION
**Pull request by jackpot51**
Before, the check view would be launched before consenting to the install and would then go immediately to the encrypt view if no error was found, not showing the user the demo mode option.

With this patch, the check view will be shown after the user has a chance to select demo mode and before encryption.

### CODE REVIEW

**Comment by tintou** [comment]
The problem is that this view can be shown at any time after the try_install_view, this change forces the user to go back to the encryption view…

**Comment by jackpot51** [comment]
I see. I can fix that. Do you agree with this statement?

Before, the check view would be launched before consenting to the install and would then go immediately to the encrypt view if no error was found, not showing the user the demo mode option.

**Comment by jackpot51** [comment]
There are all kinds of issues with the way it currently is, I will try to fix them all:

- Clicking Select on the keyboard layout screen skips the Try or Install screen
- Clicking Cancel Installation takes you back to the previous screen, rather than the Try or Install screen
- Going back to language selection, then plugging power back in will take you to the try or install screen

**Comment by jackpot51** [comment]
All three issues I found are now fixed with the latest commit.

**Comment by jackpot51** [comment]
- `previous_view` indicates which view the check view should return to when "Cancel Installation" is clicked
- `ignore_view` indicates which view to return to when "Ignore" is clicked
- If the "Ignore" button is clicked or the requirements are met, it will return to the `ignore_view`
- If the "Cancel Installation" button is clicked, it must return to the Install or Try view, and clear the `ignore_view` so that meeting the requirements will not move the installer forward in steps

File: src/Views/AbstractInstallerView.vala [hunk 1]
```diff
@@ -46,7 +48,7 @@ public abstract class AbstractInstallerView : Gtk.Grid {
         if (cancellable) {
             var cancel_button = new Gtk.Button.with_label (_("Cancel Installation"));
             cancel_button.clicked.connect (() => {
-                ((Gtk.Stack) get_parent ()).visible_child = previous_view;
```
Comment (jackpot51) [inline src/Views/AbstractInstallerView.vala]: Has this line been moved elsewhere? It will need to be called for the other views.

**Pull request merged**